peer.go

```diff
@@ -38,24 +38,57 @@ var (
 	peerRng = NewRand(time.Now().UnixNano())
 )
 
+type Connectable interface {
```

Comment:
optionally: if we can break this into a separate commit, we can PR it and submit it separately, since I think there's no behaviour changes as part of it right?